src/shared/components/oncoprint/OncoprintUtils.ts

```diff
@@ -301,6 +353,70 @@ export function makeHeatmapTracksMobxPromise(oncoprint:ResultsViewOncoprint, sam
     });
 }
 
+export function makeGenesetHeatmapExpansionsMobxPromise(oncoprint:ResultsViewOncoprint, sampleMode:boolean) {
+    return remoteData<{[genesetTrackKey: string]: IGeneHeatmapTrackSpec[]}>({
+        await: () => [
+            oncoprint.props.store.samples,
+            oncoprint.props.store.patients,
+            oncoprint.props.store.molecularProfileIdToMolecularProfile,
+            oncoprint.props.store.geneMolecularDataCache,
+            oncoprint.props.store.genesetCorrelatedGeneCache
+        ],
+        invoke: async () => {
+            const samples = oncoprint.props.store.samples.result!;
+            const patients = oncoprint.props.store.patients.result!;
+            const molecularProfileIdToMolecularProfile = oncoprint.props.store.molecularProfileIdToMolecularProfile.result!;
+            const dataCache = oncoprint.props.store.geneMolecularDataCache.result!;
+            const genesetGeneCache = oncoprint.props.store.genesetCorrelatedGeneCache.result!;
+
+            const trackGroup = oncoprint.genesetHeatmapTrackGroup;
```

Comment:
There's a new issue here after a change in PR  #982; the expansions that end up being used in the gene set heatmap tracks no longer seem to correctly observe the track group observable.

Reply:
I pushed two commits to fix the issue!

Reply:
could you please explain in more detail what the problem was and what your fix is?

Reply:
Since there's no way to represent a track group as a whole moving in the APIs for either the `<Oncoprint>` component or the OncoprintJS library, moving the gene set heatmap to a different track group index involves removing the tracks in one heatmap and adding them back in another, as per [this commit in PR #898](https://github.com/cBioPortal/cbioportal-frontend/pull/898/commits/e19c30078d4389c0b3a7a1eac8d5c69007ffc71e). At the time I wrote the expansion feature there was no reason to ever remove or add a gene set track that already has expansions, so I never wrote the `DeltaUtils` logic to update its expansions in these cases; the fix is writing that logic.

Reply:
That commit might have been easier to review if I'd moved the function declaration to a different place in a separate commit though, sorry for that.